docs/user-guide/errors.md

```diff
@@ -4,26 +4,97 @@ In addition to rule problems, Stylelint surfaces the following errors and warnin
 
 ## CSS syntax error
 
-The chosen [PostCSS syntax](usage/options.md#customsyntax) was unable to parse the source.
+The CSS parser built into Stylelint, or the chosen [custom syntax](usage/options.md#customsyntax), could not parse your code.
+
+### Unclosed block
```

Comment:
[Note] The `CSSSyntaxError` includes a message, e.g. [this demo](N4Igxg9gJgpiBcICGACYAdAdiANCAzgJ6YAuSAHguPvruBJgGYCWA5lRiAE4CuANjHzoEaAL6iQooA). I've used those messages as heading within this section so that:

- they're easier to search for
- we can give specific advice on how to resolve the issue